src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/TimeOnlyConverter.cs

```diff
@@ -3,30 +3,72 @@
 
 using System.Buffers.Text;
 using System.Diagnostics;
-using System.Globalization;
 
 namespace System.Text.Json.Serialization.Converters
 {
     internal sealed class TimeOnlyConverter : JsonConverter<TimeOnly>
     {
-        private static readonly TimeSpanConverter s_timeSpanConverter = new TimeSpanConverter();
-        private static readonly TimeSpan s_timeOnlyMaxValue = TimeOnly.MaxValue.ToTimeSpan();
+        private const int MinimumTimeOnlyFormatLength = 8; // hh:mm:ss
+        private const int MaximumTimeOnlyFormatLength = 16; // hh:mm:ss.fffffff
+        private const int MaximumEscapedTimeOnlyFormatLength = JsonConstants.MaxExpansionFactorWhileEscaping * MaximumTimeOnlyFormatLength;
 
         public override TimeOnly Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
         {
-            TimeSpan timespan = s_timeSpanConverter.Read(ref reader, typeToConvert, options);
+            if (reader.TokenType != JsonTokenType.String)
+            {
+                ThrowHelper.ThrowInvalidOperationException_ExpectedString(reader.TokenType);
+            }
 
-            if (timespan < TimeSpan.Zero || timespan > s_timeOnlyMaxValue)
+            if (!JsonHelpers.IsInRangeInclusive(reader.ValueLength, MinimumTimeOnlyFormatLength, MaximumEscapedTimeOnlyFormatLength))
             {
-                ThrowHelper.ThrowJsonException();
+                ThrowHelper.ThrowFormatException(DataType.TimeOnly);
             }
 
+            ReadOnlySpan<byte> source = stackalloc byte[0];
+            if (!reader.HasValueSequence && !reader.ValueIsEscaped)
+            {
+                source = reader.ValueSpan;
+            }
+            else
+            {
+                Span<byte> stackSpan = stackalloc byte[MaximumEscapedTimeOnlyFormatLength];
+                int bytesWritten = reader.CopyString(stackSpan);
+                source = stackSpan.Slice(0, bytesWritten);
+            }
+
+            byte firstChar = source[0];
+            int firstSeparator = source.IndexOfAny((byte)'.', (byte)':');
+            if (!JsonHelpers.IsDigit(firstChar) || firstSeparator < 0 || source[firstSeparator] == (byte)'.')
```

Comment:
>firstSeparator < 0 || source[firstSeparator] == (byte)'.' [](http://example.com/codeflow?start=51&length=57)

would it be better to strictly check positions like 

[2] == ':' && [5] == ':' && (length < 8 || [9] == '.')?

I am asking this because I don't know if Utf8Parser.TryParse can succeed when having something like "1:2:3.1234567"

or we consider this is a valid format for S.T.J?